OsmAnd/src-google/net/osmand/plus/inapp/InAppPurchaseHelperImpl.java

```diff
@@ -532,59 +528,168 @@ private PurchaseInfo getPurchaseInfo(Purchase purchase) {
+										introductoryPriceAmountMicros, introductoryPricePeriod, introductoryPriceCycles));
+							} catch (ParseException e) {
+								LOG.error(e);
+							}
+						}
+					}
 				}
 			}
+		} else {
+			ProductDetails.OneTimePurchaseOfferDetails purchaseOfferDetails = productDetails.getOneTimePurchaseOfferDetails();
+			if (purchaseOfferDetails != null) {
+				inAppPurchase.setPrice(purchaseOfferDetails.getFormattedPrice());
+				inAppPurchase.setOriginalPrice(purchaseOfferDetails.getFormattedPrice());
+				inAppPurchase.setPriceCurrencyCode(purchaseOfferDetails.getPriceCurrencyCode());
+			}
+		}
+	}
+
+
+	// equivalent to skuDetails.originalPriceAmountMicros
+	private long getPriceAmountMicros(ProductDetails productDetails) {
+		ProductDetails.PricingPhase phase = getMostPricedOfferPhase(productDetails);
+		long price = 0;
+		if (phase != null) {
+			price = phase.getPriceAmountMicros();
 		}
-		if (inAppPurchase instanceof InAppSubscription) {
-			InAppSubscription s = (InAppSubscription) inAppPurchase;
-			s.restoreState(ctx);
-			s.restoreExpireTime(ctx);
-			SubscriptionStateHolder stateHolder = subscriptionStateMap.get(s.getSku());
-			if (stateHolder != null) {
-				s.setState(ctx, stateHolder.state);
-				s.setExpireTime(ctx, stateHolder.expireTime);
+		return price;
+	}
+
+	// skuDetails.subscriptionPeriod
+	@Nullable
+	private String getBillingPeriod(ProductDetails productDetails) {
+		ProductDetails.PricingPhase phase = getMostPricedOfferPhase(productDetails);
+		String period = null;
+		if (phase != null) {
+			period = phase.getBillingPeriod();
+		}
+		return period;
+	}
+
+	// skuDetails.priceCurrencyCode
+	@Nullable
+	private String getPriceCurrencyCode(ProductDetails productDetails) {
+		ProductDetails.PricingPhase phase = getMostPricedOfferPhase(productDetails);
+		String period = null;
+		if (phase != null) {
+			period = phase.getPriceCurrencyCode();
+		}
+		return period;
+	}
+
+
+	// skuDetails.freeTrialPeriod
+	private Period getTrialDays(ProductDetails productDetails) {
+		String billingPeriod = getLeastPricedBillingPeriod(productDetails);
+		Period trialPeriod = null;
+		if (billingPeriod != null) {
+			try {
+				trialPeriod = Period.parse(billingPeriod);
+			} catch (ParseException error) {
+				LOG.error("Can't parse " + billingPeriod + "period");
 			}
-			if (s.getState().isGone() && s.hasStateChanged()) {
-				ctx.getSettings().LIVE_UPDATES_EXPIRED_FIRST_DLG_SHOWN_TIME.set(0L);
-				ctx.getSettings().LIVE_UPDATES_EXPIRED_SECOND_DLG_SHOWN_TIME.set(0L);
+		}
+		return trialPeriod;
+	}
+
+	@Nullable
+	private String getLeastPricedBillingPeriod(ProductDetails productDetails) {
+		ProductDetails.PricingPhase pricingPhase = getLeastPricedOfferPhase(productDetails);
+		if (pricingPhase != null) {
+			return pricingPhase.getBillingPeriod();
+		} else {
+			return null;
+		}
+	}
+
+	@Nullable
+	private ProductDetails.PricingPhase getMostPricedOfferPhase(ProductDetails productDetails) {
```

Comment:
It is not clear - idea with most/least priced logic. Does it discussed with someone? Why there no any mention about base plan?